centreon/src/Core/HostSeverity/Application/UseCase/AddHostSeverity/AddHostSeverity.php

```diff
@@ -0,0 +1,145 @@
+use Core\HostSeverity\Application\Exception\HostSeverityException;
+use Core\HostSeverity\Application\Repository\ReadHostSeverityRepositoryInterface;
+use Core\HostSeverity\Application\Repository\WriteHostSeverityRepositoryInterface;
+use Core\HostSeverity\Domain\Model\HostSeverity;
+use Core\HostSeverity\Domain\Model\NewHostSeverity;
+use Core\HostSeverity\Infrastructure\API\AddHostSeverity\AddHostSeverityPresenter;
+use Core\ViewImg\Application\Repository\ReadViewImgRepositoryInterface;
+
+final class AddHostSeverity
+{
+    use LoggerTrait;
+
+    public function __construct(
+        private readonly WriteHostSeverityRepositoryInterface $writeHostSeverityRepository,
+        private readonly ReadHostSeverityRepositoryInterface $readHostSeverityRepository,
+        private readonly ReadViewImgRepositoryInterface $readViewImgRepository,
+        private readonly ContactInterface $user
+    ) {
+    }
+
+    /**
+     * @param AddHostSeverityRequest $request
+     * @param AddHostSeverityPresenter $presenter
+     */
+    public function __invoke(AddHostSeverityRequest $request, PresenterInterface $presenter): void
+    {
+        try {
+            if (! $this->user->hasTopologyRole(Contact::ROLE_CONFIGURATION_HOSTS_CATEGORIES_READ_WRITE)) {
+                $this->error(
+                    "User doesn't have sufficient rights to add host severities",
+                    ['user_id' => $this->user->getId()]
+                );
+                $presenter->setResponseStatus(
+                    new ForbiddenResponse(HostSeverityException::addNotAllowed()->getMessage())
+                );
+            } elseif ($this->readHostSeverityRepository->existsByName(trim($request->name))) {
+                $this->error(
+                    'Host severity name already exists',
+                    ['hostseverity_name' => trim($request->name)]
+                );
+                $presenter->setResponseStatus(
+                    new ConflictResponse(HostSeverityException::hostNameAlreadyExists())
+                );
+            } elseif (
+                0 === $request->iconId
+                || ! $this->readViewImgRepository->existsOne($request->iconId)
+            ) {
+                $this->error(
+                    'Host severity icon does not exist',
+                    ['hostseverity_name' => trim($request->name)]
+                );
+                $presenter->setResponseStatus(
+                    new ConflictResponse(HostSeverityException::iconDoesNotExist($request->iconId))
+                );
+            } else {
+                $newHostSeverity = new NewHostSeverity(
+                    trim($request->name),
+                    trim($request->alias),
+                    $request->level,
+                    $request->iconId,
+                );
+                $newHostSeverity->setActivated($request->isActivated);
+                $newHostSeverity->setComment($request->comment ? trim($request->comment) : null);
```

Comment:
```suggestion
                $newHostSeverity->setComment($request->comment ?$request->comment : null);
```
Same remark as above

Reply:
✅ done with added tests